src/pspm_extract_segments.m

```diff
@@ -1,680 +1,738 @@
-elseif ~isempty(options.outputfile) && ~ischar(options.outputfile)
-    warning('ID:invalid_input', 'options.outputfile has to be a string.'); return;
-elseif ~isnumeric(options.overwrite) && ~islogical(options.overwrite)
-    warning('ID:invalid_input', 'Options.overwrite has to be numeric or logical.'); return;
-elseif ~isnumeric(options.dont_ask_overwrite) && ~islogical(options.dont_ask_overwrite)
-    warning('ID:invalid_input', 'Options.dont_ask_overwrite has to be numeric or logical.'); return;
-elseif strcmpi(options.timeunit, 'markers') && ~all(size(data_fn) == size(options.marker_chan))
-    warning('ID:invalid_input', '''data_fn'' and ''options.marker_chan'' do not have the same size.'); return;
-elseif (manual_chosen ==1) ||  strcmpi(model_strc.modeltype,'glm')
-    if any(cellfun(@(x) ~strcmpi(x, 'marker') && ~isnumeric(x), options.marker_chan))
-        warning('ID:invalid_input', 'Options.marker_chan has to be numeric or ''marker''.'); return;
-    elseif strcmpi(options.timeunit, 'markers') ...
-            && any(cellfun(@(x) isnumeric(x) && x <= 0, options.marker_chan))
-        warning('ID:invalid_input', ['''markers'' specified as a timeunit but ', ...
-            'no valid marker channel is defined.']); return;
+    
+    if ~isstruct(options)
+        warning('ID:invalid_input', 'Options must be a struct.'); return;
+    end;
+    
+    % set default timeunit
+    if ~isfield(options, 'timeunit')
+        options.timeunit = 'seconds';
+    else
+        options.timeunit = lower(options.timeunit);
+    end;
+    
+    % set default marker_chan, if it is a glm struct (only for non-raw data)
+    if manual_chosen == 1 || (manual_chosen == 0 && strcmpi(model_strc.modeltype,'glm'))
+        if ~isfield(options, 'marker_chan')
+            options.marker_chan = repmat({-1}, numel(data_fn),1);
+        elseif ~iscell(options.marker_chan)
+            options.marker_chan = repmat({options.marker_chan}, size(data_fn));
+        end;
     end;
-end;
-
-if manual_chosen == 1 || strcmpi(model_strc.modeltype, 'glm')
-    n_sessions = numel(data_fn);
-else
-    n_sessions = numel(model_strc.input.scr);
-end
-
-% load timing
-if manual_chosen == 1
-    [~, multi]  = pspm_get_timing('onsets', timing, options.timeunit);
-    % If the timeunit is markers, the multi struct holds for every session.
-    % Thus we need as many replicas as there are sessions
-    if strcmpi(options.timeunit, 'markers')
-        temp = multi;
-        clear multi;
-        for k=1:n_sessions
-            multi(k) = temp;
+    
+    % set default length
+    if ~isfield(options, 'length')
+        options.length = -1;
+    end;
+    
+    % default plot
+    if ~isfield(options, 'plot')
+        options.plot = 0;
+    end;
+    
+    % outputfile
+    if ~isfield(options, 'outputfile')
+        options.outputfile = '';
+    end;
+    
+    % overwrite
+    if ~isfield(options, 'overwrite')
+        options.overwrite = 0;
+    end;
+    
+    % dont_ask_overwrite

```

Comment:
(Low priority) Now I see that we have this undocumented previously implemented "don't ask overwrite" option for some reason, and we don't need it. Could you remove everything related to this option and change line 715 to `else`?

Reply:
Copy that